conan/tools/qbs/qbsdeps.py

```diff
@@ -0,0 +1,161 @@
+
+            def _get_name_with_namespace(namespace, name):
+                """
+                Build a name with a namespace, e.g., openssl-crypto
+                """
+                return f"{namespace}-{name}"
+
+            def get_components(dep):
+                ret = {}
+                for comp_ref_name, info in dep.cpp_info.get_sorted_components().items():
+                    comp_name = _get_component_name(dep, comp_ref_name)
+                    ret[comp_name] = info
+                return ret
+
+            # copy & paste from pkgconfig
+            # TODO: versions
+            def get_cpp_info_requires_names(dep, cpp_info):
+                ret = []
+                dep_ref_name = dep.ref.name
+                for req in cpp_info.requires:
+                    pkg_ref_name, comp_ref_name = (
+                        req.split("::") if "::" in req else (dep_ref_name, req)
+                    )
+
+                    if dep_ref_name != pkg_ref_name:
+                        try:
+                            req_conanfile = transitive_reqs[pkg_ref_name]
+                        except KeyError:
+                            # If the dependency is not in the transitive, might be skipped
+                            continue
+                    # For instance, dep == "hello/1.0" and req == "hello::cmp1" -> hello == hello
+                    else:
+                        req_conanfile = dep
+
+                    comp_name = _get_component_name(req_conanfile, comp_ref_name)
+                    if not comp_name:
+                        pkg_name = _get_package_name(req_conanfile)
+                        # Creating a component name with namespace, e.g., dep-comp1
+                        comp_name = _get_name_with_namespace(pkg_name, comp_ref_name)
+                    ret.append((comp_name, req_conanfile.ref.version))
+                return ret
+
+            if not dep.cpp_info.has_components:
+                moduleName = _get_package_name(dep)
+                requires = get_cpp_info_requires_names(dep, dep.cpp_info._package)
+                if not requires:
+                    # If no requires were found, let's try to get all the direct visible
+                    # dependencies, e.g., requires = "other_pkg/1.0"
+                    for deprequire, _ in dep.dependencies.direct_host.items():
+                        requires.append((deprequire.ref.name, deprequire.ref.version))
+                file = _QbsDepsModuleFile(self, dep, dep.cpp_info._package, requires, moduleName)
+                qbs_files.append(file)
+            else:
+                full_requires = []
+                for moduleName, component in get_components(dep).items():
+                    requires = get_cpp_info_requires_names(dep, component)
+                    qbs_files.append(_QbsDepsModuleFile(self, dep, component, requires, moduleName))
+                    full_requires.append((moduleName, dep.ref.version))
```

Comment:
```suggestion
                for module_name, component in get_components(dep).items():
                    requires = get_cpp_info_requires_names(dep, component)
                    qbs_files.append(_QbsDepsModuleFile(self, dep, component, requires, module_name))
                    full_requires.append((module_name, dep.ref.version))
```
Python variable style if possible, please.